dwave/gate/circuit.py

```diff
@@ -61,6 +62,7 @@
     from dwave.gate.operations.base import Operation
 
 Qubits = Union[Qubit, Sequence[Qubit]]
+Bits = Union[Bit, Sequence[Bit]]
```

Comment:
I find this confusing. Imagine `list = element | list`.

You can't say `for bit in bits`, because it will sometimes fail and sometimes work.

Reply:
I can definitely see that as confusing. A lot of places take either a (qu)bit or a sequence of (qu)bits, so this made things a bit cleaner, but I get your point. Wouldn't mind either renaming it e.g., `BitOrBits` or just using `Union[Bit, Sequence[Bit]]`, although I personally found this clear enough.

Part of me would have preferred just having either `Bit` _or_ `Sequence[Bit]` accepted, instead of in some cases supporting both. That might actually be the better solution long-term.

Reply:
An elegant way would be to use a List monad (or sequence monad) for containing non-determinism of (qu)bit count here. (But that's not really an option in Python)

I guess the second best (for clarity) is to accept element _or_ sequence explicitly (single dispatch / pattern matching?).

But I definitely get the appeal of accepting `Union` and just dealing with it, so that user doesn't have to. I guess that's the "batteries included" approach, which is fine. It's just confusing if you have to deal with such a `Union` result.